internal/cmd/launcher.go

```diff
@@ -225,6 +233,21 @@ func k6buildProvision(gs *state.GlobalState, deps k6deps.Dependencies) (commandE
 	return &customBinary{binary.Path}, nil
 }
 
+func getBinaryCacheDir(fs fsext.Fs) (string, error) {
+	// nolint:forbidigo
+	// TODO: init this as part of the GlobalState
+	// and remote it from here
+	userdir, err := os.UserCacheDir()
+	if err != nil {
+		return "", err
+	}
```

Comment:
I intend to address this having a field in `GlobalState` but I want to minimize the changes requires on this pull request.

The next week, I will open a separated pull request that I intend to merge only after we have merged Binary Provisioning.

Reply:
FYI: Here is a draft with that approach https://github.com/grafana/k6/pull/4743